### DISCUSSION TITLE
Proplem on installation

### DISCUSSION THREAD

**Discussion by brl1999** [materialsproject/pymatgen#4152] [Q&A] [answered]
Hi. I met a problem after the installation.  The pip list suggests that I've installed the pymatgen 2020.8.9 sucessfully, but when I tried the basic code `from mp_api.client import MPRester` , it turned out with the error:
`ImportError                               Traceback (most recent call last)
Cell In[3], line 1
----> 1 from mp_api.client import MPRester

File ~\.conda\envs\env\lib\site-packages\mp_api\client\__init__.py:8
      5 from importlib.metadata import PackageNotFoundError, version
      7 from .core import MPRestError
----> 8 from .mprester import MPRester
     10 try:
     11     __version__ = version("mp_api")

File ~\.conda\envs\env\lib\site-packages\mp_api\client\mprester.py:11
      8 from json import loads
      9 from typing import Literal
---> 11 from emmet.core.electronic_structure import BSPathType
     12 from emmet.core.mpid import MPID
     13 from emmet.core.settings import EmmetSettings

File ~\.conda\envs\env\lib\site-packages\emmet\core\electronic_structure.py:26
     23 from pymatgen.symmetry.bandstructure import HighSymmKpath
     24 from typing_extensions import Literal
---> 26 from emmet.core.material_property import PropertyDoc
     27 from emmet.core.mpid import MPID
     28 from emmet.core.settings import EmmetSettings

File ~\.conda\envs\env\lib\site-packages\emmet\core\material_property.py:10
      7 from pymatgen.core import Structure
      8 from datetime import datetime
---> 10 from emmet.core.material import PropertyOrigin
     11 from emmet.core.mpid import MPID
     12 from emmet.core.structure import StructureMetadata

File ~\.conda\envs\env\lib\site-packages\emmet\core\material.py:13
     10 from pymatgen.core.structure import Molecule
     12 from emmet.core.mpid import MPID, MPculeID
---> 13 from emmet.core.structure import MoleculeMetadata, StructureMetadata
     14 from emmet.core.vasp.validation import DeprecationMessage
     15 from emmet.core.common import convert_datetime

File ~\.conda\envs\env\lib\site-packages\emmet\core\structure.py:13
     11 from emmet.core.utils import get_graph_hash
     12 from emmet.core.base import EmmetBaseModel
---> 13 from emmet.core.symmetry import PointGroupData, SymmetryData
     15 T = TypeVar("T", bound="StructureMetadata")
     16 S = TypeVar("S", bound="MoleculeMetadata")

File ~\.conda\envs\env\lib\site-packages\emmet\core\symmetry.py:6
      4 from pymatgen.core import Structure
      5 from pymatgen.core.structure import Molecule
----> 6 from pymatgen.symmetry.analyzer import (
      7     PointGroupAnalyzer,
      8     SpacegroupAnalyzer,
      9     spglib,
     10     SymmetryUndeterminedError,
     11 )
     13 from emmet.core.settings import EmmetSettings
     14 from emmet.core.utils import ValueEnum

ImportError: cannot import name 'SymmetryUndeterminedError' from 'pymatgen.symmetry.analyzer' (C:\Users\brl1999\.conda\envs\env\lib\site-packages\pymatgen\symmetry\analyzer.py)`
My python edition is 3.9. I wonder how to fix the problem. Thanks for the suggestions.

**Comment by JaGeo** [comment, marked as answer]
Python 3.9 is not supported by pymatgen anymore. Please switch to Python 3.10 and repeat the installation. Thank you 😀.

**Reply by brl1999** [reply]
Thank you very much! The problem has been solved.

**Comment by amkrajewski** [comment]
I've run into a similar issue when installing `pymatgen` alongside other tools in Python 3.11 and 3.12.

@JaGeo It appears that `emmet` is [currently](https://github.com/materialsproject/emmet/blob/644b13713c5cad821acec63d12f3e5c9954b579c/emmet-core/setup.py#L32) underspecifying its dependency on  `pymatgen`. It requires `>=2024.6.10` while [this "linting" commit](https://github.com/materialsproject/pymatgen/commit/f6b4073add07a02f5726fe348f49f2e5100e3c8e) changing name from `SymmetryUndetermined` to `SymmetryUndeterminedError` is expected to be present; thus, dependency should be `pymatgen>=2024.9.10`.

@brl1999 If you need to use Python 3.9 for some reason, try `pip install emmet-core==0.84.0`.

**Reply by brl1999** [reply]
Thank you very much! The problem has been solved.
